networkx/utils/configs.py

```diff
@@ -291,11 +298,28 @@ def _check_config(self, key, value):
         elif key == "cache_converted_graphs":
             if not isinstance(value, bool):
                 raise TypeError(f"{key!r} config must be True or False; got {value!r}")
+        elif key == "warnings":
+            if not (isinstance(value, set) and all(isinstance(x, str) for x in value)):
+                raise TypeError(
+                    f"{key!r} config must be a set of warning names; got {value!r}"
+                )
+            known_warnings = {"cache"}
+            if missing := {x for x in value if x not in known_warnings}:
+                missing = ", ".join(map(repr, sorted(missing)))
+                raise ValueError(
+                    f"Unknown warning when setting {key!r}: {missing}. Valid entries: "
+                    + ", ".join(sorted(known_warnings))
+                )
 
 
 # Backend configuration will be updated in backends.py
 config = NetworkXConfig(
     backend_priority=[],
     backends=Config(),
     cache_converted_graphs=bool(os.environ.get("NETWORKX_CACHE_CONVERTED_GRAPHS", "")),
+    warnings={
+        x.strip()
+        for x in os.environ.get("NETWORKX_WARNINGS", "cache").split(",")
```

Comment:
Is this a new env var?  I didn't notice it documented anywhere.  Am I interpreting it right that you add a comma-separated list of "things" to ignore warnings for?  If so, it seems like the name should be something more like `NETWORK_WARNINGS_TO_DISCARD` or something like that but better.

Reply:
Looks like I misinterpreted this and `NETWORKX_WARNINGS` is a list of things/categories for which warnings *should* be shown.  I'm assuming this is then used to override configuration settings that have certain warnings disabled? 

Either way, can you also update the docstring for `NetworkXConfig` to include this new env var to the list of existing env vars [here](https://github.com/networkx/networkx/blob/607ed7e783af974ceba62e493038ce7d668cae76/networkx/utils/configs.py#L256-L259)?

Reply:
Yes, the set is positive, not negative. Do you think we should change it to be negative?

The docstring is already updated. Do you mean to update docs somewhere else?

Reply:
What about something like `NETWORKX_WARNINGS_TO_IGNORE` and/or `nx.config.warnings_to_ignore`? Is this more convenient to use? Would we also want to keep the current `nx.config.warnings` and to keep `warnings` and `warnings_to_ignore` consistent such that `warnings ^ warnings_to_ignore == set()` and `warnings | warnings_to_ignore == all_warnings`? When using environment variables, I do think choosing which ones to ignore is more convenient, but from Python I prefer `nx.config.warnings`.

Alternatively, is there a good name for a boolean config to enable/disable the specific warning about caching (i.e., to not use a more generic `nx.config.warning` approach)?